From 25b7c389d68d2fb3bde470b7ec4bf9cf7271d60f Mon Sep 17 00:00:00 2001
From: Patrick McCarty <patrick.mccarty@intel.com>
Date: Fri, 16 Nov 2018 23:34:15 -0800
Subject: [PATCH 2/2] Fix compilation when --disable-des is set

Fixes this compiler error:

  scapi.c: In function 'sc_encrypt':
  scapi.c:1254:5: error: 'pad_size' undeclared (first use in this function)
       pad_size = pai->pad_size;
       ^~~~~~~~

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 snmplib/scapi.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/snmplib/scapi.c b/snmplib/scapi.c
index 8ad1d70..81537d3 100644
--- a/snmplib/scapi.c
+++ b/snmplib/scapi.c
@@ -1251,11 +1251,11 @@ sc_encrypt(const oid * privtype, size_t privtypelen,
 
         QUITFUN(SNMPERR_GENERR, sc_encrypt_quit);
     }
-    pad_size = pai->pad_size;
-
     memset(my_iv, 0, sizeof(my_iv));
 
 #ifndef NETSNMP_DISABLE_DES
+    pad_size = pai->pad_size;
+
     if (USM_CREATE_USER_PRIV_DES == (pai->type & USM_PRIV_MASK_ALG)) {
 
         /*
-- 
2.19.1

